rfd/0235-desktop-access-linux.md

```diff
@@ -0,0 +1,138 @@
+
+Unix socket created by `Xvfb` will be secured using `Xauthority` file generated by Teleport and shared between `Xvfb`,
+desktop environment and Linux Desktop Service. `Xvfb` will not create any TCP sockets.
+
+### Frame encoding
+
+Modified regions will be discovered using [DAMAGE](https://www.x.org/archive/X11R7.5/doc/damageproto/damageproto.txt)
+extension. Each region will be encoded using [QOI](https://en.wikipedia.org/wiki/QOI_(image_format)) + [zSTD](https://github.com/facebook/zstd)
+combination using Rust encoder to maximize performance. This encoding is supported by IronRDP in `qoiz` feature, so
+we can leverage their decoder in UI.
+
+### Screen resize
+
+Screen will be resized using [XRANDR](https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/raw/master/randrproto.txt?ref_type=heads)
+extension. `Xvfb` will be started using maximum supported screen size (8192x8192) and immediately resized down to 
+requested size. This is needed as `Xvfb` won't allow resizes to size bigger than the initial one.
+
+### Mouse and keyboard input
+
+Mouse and keyboard events will be translated into [XTEST](https://www.x.org/releases/X11R7.7-RC1/doc/xextproto/xtest.html)
+extensions call. It supports all currently supported events.
+
+Information about pointer changes will be obtained using [XFIXES](https://cgit.freedesktop.org/xorg/proto/fixesproto/plain/fixesproto.txt)
+extension: `CursorNotify` and `GetCursorImage`
+
+### Clipboard sharing
+
+Clipboard will be shared by monitoring and managing `CLIPBOARD` selection. Data will be copied both way as soon as
+it is available, the same way we do it for Windows Desktop. The message flow and between UI and Linux Desktop Service
+and permission model will be the same as defined in RFD 49 for Windows Desktop.
+
+### Directory sharing
+   
+FUSE file system will be created using [go-fuse](https://github.com/hanwen/go-fuse). It will be mounted in user's home 
+directory, and it will redirect and translate all requests to TDPB messages that will be handled by UI.
+
+If FUSE is not available on Linux host directory sharing will be disabled and warning will be emitted
+
+### Concurrent sessions
+
+Each login will start separate session. This is in contrast to how Windows Desktop Access works, as it will reuse
+existing session on log out current user, but it's similar to how SSH session works.
+
+### Session recordings
+
+Sessions will be recorded in the same way as they do for Windows desktops. Playback will reuse most of the code as well.
+
+### Configuration
+
+New `teleport.yaml` section for `linux_desktop_service`:
+
+```yaml
+linux_desktop_service:
+  enabled: yes # default false
+  listen_addr: 0.0.0.0:3029
+  public_addr: linux.desktop.example.com:3080
+  # optional, xsessions will provide regexes for filtering available sessions to present in UI
+  xsessions:
+    included: "^Xfce.*" # defaults to ^.*$
+    excluded: ".*restricted" #defualts to no excludes
```

Comment:
What if a session matches both included & excluded?

What if it matches neither?